lib/player.js

```diff
@@ -3381,6 +3431,35 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
         .set('metadataType', metadataType)
         .set('payload', payload);
     this.dispatchEvent(shaka.Player.makeEvent_(eventName, data));
+
+    this.addMetadataToRegionTimeline_(
+        startTime, endTime, metadataType, payload);
+  }
+
+
+  /**
+   * Add metadata to region timeline
+   *
+   * @param {number} startTime
+   * @param {?number} endTime
+   * @param {string} metadataType
+   * @param {shaka.extern.MetadataFrame} payload
+   * @private
+   */
+  addMetadataToRegionTimeline_(startTime, endTime, metadataType, payload) {
+    if (!this.metadataRegionTimeline_) {
+      return;
+    }
+    /** @type {shaka.extern.MetadataTimelineRegionInfo} */
+    const region = {
+      schemeIdUri: metadataType,
+      startTime,
+      endTime: endTime || Infinity,
+      id: '',
+      payload,
+    };
+    region.id = shaka.util.StringUtils.hash(JSON.stringify(region));
```

Comment:
The hash seems unnecessary.  Could this ID be composed from the metadata type + start + end instead?

Reply:
You can have multiple ID3s with those same values... it is not valid to have that.